docker/sagemaker/Dockerfile.endpoint

```diff
@@ -0,0 +1,75 @@
+# Docker file for building a docker image for running GraphStorm code on Amazon SageMaker
+# Note: Distributed graph partition will use another docker image which will come soon.
+
+ARG DEVICE=gpu
+ARG SOURCE
+
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker} as branch-gpu
+ENV dev_type=GPU
+ARG TORCH_VERSION
+
+# Uninstall preset Pytorch packages
+RUN pip3 uninstall -y torchdata torch torchvision torchaudio
+
+# Install Dependencies to fit DGL GraphBolt requirements
+RUN pip3 install \
+        torchdata==0.9.0 \
+        pydantic==2.7.1 \
+        transformers==4.47.1 \
+        ogb==1.3.6 \
+    && rm -rf /root/.cache
+
+RUN pip3 install \
+    torch==2.3 torchvision torchaudio \
+    --index-url https://download.pytorch.org/whl/cu121 \
+    && rm -rf /root/.cache
+
+# Install DGL GPU version
+RUN pip3 install dgl==2.5.0+cu121 -f https://data.dgl.ai/wheels-internal/torch-2.3/cu121/repo.html \
+    && rm -rf /root/.cache
+
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.3.0-cpu-py311-ubuntu20.04-sagemaker} as branch-cpu
+ENV dev_type=CPU
+ARG TORCH_VERSION
+
+# Uninstall preset Pytorch packages
+RUN pip3 uninstall -y torchdata torch torchvision torchaudio
+
+# Install Dependencies to fit DGL GraphBolt requirements
+RUN pip3 install \
+        torchdata==0.9.0 \
+        pydantic==2.7.1 \
+        transformers==4.47.1 \
+        ogb==1.3.6 \
+    && rm -rf /root/.cache
+
+RUN pip3 install \
+    torch==2.3 torchvision torchaudio \
+    --index-url https://download.pytorch.org/whl/cpu \
+    && rm -rf /root/.cache
+
+# Install DGL CPU version
+RUN pip3 install dgl==2.5.0 -f https://data.dgl.ai/wheels-internal/torch-2.3/repo.html \
+    && rm -rf /root/.cache
+
+FROM branch-${DEVICE} AS final
+
+LABEL maintainer="Amazon AI Graph ML team"
+
+# Install MPI etc needed by DistDGL
+RUN apt-get update; apt-get install -y --no-install-recommends libopenmpi-dev \
+    build-essential software-properties-common; add-apt-repository ppa:ubuntu-toolchain-r/test; \
+    apt-get update; apt-get upgrade libstdc++6 -y && rm -rf /var/lib/apt/lists/*
+
+# /opt/ml and all subdirectories are utilized by SageMaker, we use the /code subdirectory to store our user code.
+# TODO(xiangsx): change to pip install when PIP package is available
+COPY code/graphstorm/ /opt/ml/code/graphstorm/
+ENV PYTHONPATH="/opt/ml/code/graphstorm/python/:${PYTHONPATH}"
+RUN cp /opt/ml/code/graphstorm/sagemaker/run/* /opt/ml/code/
```

Comment:
We may want to avoid using /opt/ml/code here, see https://github.com/awslabs/graphstorm/blob/d6a928b012c9076acbbe8867eefe7ab0468060c8/docker/sagemaker/Dockerfile.sm#L50-L53

Reply:
For inference, we don't need copy the  sagemaker/run code.

Reply:
But we will need `sagemaker/launch/realtime_entry_points/`?

Reply:
Cannot do that as the entry point should be in a specific position, i.e., `/opt/ml/model/code/`. But during deployment SageMaker will attach an external folder to it, hence cleaning any contents backed in the Docker image.